services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipInvitationManagerImpl.java

```diff
@@ -162,6 +192,49 @@ public Count getOpenInvitationCountForUser(String principalId) {
 		return result;
 	}
 
+	@Override
+	public InviteeVerificationSignedToken verifyInvitee(Long userId, String membershipInvitationId) {
+		ValidateArgument.required(userId, "userId");
+		ValidateArgument.required(membershipInvitationId, "membershipInvitationId");
+		// Get list of email addresses associated with userId
+		List<PrincipalAlias> aliases = principalAliasDAO.listPrincipalAliases(userId);
```

Comment:
instead of retrieving all aliases, call a DAO-level method that just checks whether the given user has the alias of interest